File: app/src/main/java/com/parassidhu/popularmovies/adapters/ReviewAdapter.java
package com.parassidhu.popularmovies.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.parassidhu.popularmovies.R;
import com.parassidhu.popularmovies.models.ReviewItem;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ReviewAdapter extends RecyclerView.Adapter {
    private Context context;
    private List<ReviewItem> reviewItems;

    public ReviewAdapter(Context context, List<ReviewItem> reviewItems) {
        this.context = context;
        this.reviewItems = reviewItems;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = View.inflate(context, R.layout.row_review, null);
        return new ReviewViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder viewHolder, int i) {
        ((ReviewViewHolder) viewHolder).bind(i);
    }

    public List<ReviewItem> getReviewItems() {
        return reviewItems;
    }

    public void setReviewItems(List<ReviewItem> reviewItems) {
        this.reviewItems = reviewItems;
        notifyDataSetChanged();
    }

    @Override
    public int getItemCount() {
        return reviewItems.size();
    }

    class ReviewViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.username) TextView username;
        @BindView(R.id.review) TextView review;

        ReviewViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        void bind(int position){
            ReviewItem item = reviewItems.get(position);
            username.setText(item.getAuthor());
            review.setText(item.getContent());

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    if (review.getMaxLines()==3){
                        review.setMaxLines(200);
                    } else
                        review.setMaxLines(3);
                }
            });
        }
    }
}

File: app/src/main/java/com/parassidhu/popularmovies/adapters/ReviewAdapter.kt
package com.parassidhu.popularmovies.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.parassidhu.popularmovies.R
import com.parassidhu.popularmovies.models.ReviewItem

import java.util.ArrayList

import butterknife.BindView
import butterknife.ButterKnife

class ReviewAdapter(
        private val context: Context,
        private var reviewItems: List<ReviewItem>?
) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): RecyclerView.ViewHolder {
        val view = View.inflate(context, R.layout.row_review, null)
        return ReviewViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: RecyclerView.ViewHolder, i: Int) {
        (viewHolder as ReviewViewHolder).bind(i)
    }

    fun getReviewItems(): List<ReviewItem>? {
        return reviewItems
    }

    fun setReviewItems(reviewItems: List<ReviewItem>) {
        this.reviewItems = reviewItems
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return reviewItems!!.size
    }

    internal inner class ReviewViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        @BindView(R.id.username)
        @JvmField
        var username: TextView? = null

        @BindView(R.id.review)
        @JvmField
        var review: TextView? = null

        init {
            ButterKnife.bind(this, itemView)
        }

        fun bind(position: Int) {
            val (author, content) = reviewItems!![position]
            username!!.text = author
            review!!.text = content

            itemView.setOnClickListener {
                if (review!!.maxLines == 3) {
                    review!!.maxLines = 200
                } else
                    review!!.maxLines = 3
            }
        }
    }
}
